File: python/nt_solutions/segmentation_2_snakes_param/exo6.py
G = grad(W);
G = G[:,:,0] + 1j*G[:,:,1]
EvalG = lambda gamma: bilinear_interpolate(G, imag(gamma), real(gamma))
EvalW = lambda gamma: bilinear_interpolate(W, imag(gamma), real(gamma))
#
gamma = gamma0;
displist = around(linspace(0,niter,10))
k = 0;
clf
imageplot(transpose(f))
for i in arange(0,niter+1):
    N = normal(gamma)
    g = EvalW(gamma) * normalC(gamma) - dotp(EvalG(gamma), N) * N
    gamma = gamma + dt*g
    gamma = resample( gamma )
    # impose start/end point
    gamma[0] = x0
    gamma[-1] = x1
    if i==displist[k]:   
        lw = 1;    
        if i==0 or i==niter:
            lw = 4;
        cplot(gamma, 'r', lw);
        k = k+1;
        axis('equal'); axis('off');
        plot(real(gamma[0]), imag(gamma[0]), 'b.', markersize=20)
        plot(real(gamma[-1]), imag(gamma[-1]), 'b.', markersize=20);

File: r/nt_solutions/segmentation_2_snakes_param/exo6.R




G <- grad(W)
G <- G[,,1] + 1i*G[,,2]
EvalG <- function(gamma){ bilinear_interpolate(G, Im(gamma), Re(gamma)) }
EvalW <- function(gamma){ bilinear_interpolate(W, Im(gamma), Re(gamma)) }
#
gamma <- gamma0
displist <- round(seq(1, niter, length=10))
k <- 1
imageplot(t(f))

for (i in 1:niter){
  N <- normal(gamma)
  g <- EvalW(gamma)*normalC(gamma) - N*dotp(EvalG(gamma), N)
  gamma <- gamma + dt*g
  gamma <- resample( gamma )
  # impose start/end point
  gamma[1] <- x0
  gamma[length(gamma)] <- x1
  if (i==displist[k]){
    lw <- 1
    if (i==1 || i==niter){
      lw <- 4
    }
    cplot(gamma, type="l", pch=1, lw=lw, lty=1, col="red", add=TRUE)
    k <- k+1
    points(Re(gamma[1]), Im(gamma[1]), pch=16, cex=1.5, col="blue")
    points(Re(gamma[length(gamma)]), Im(gamma[length(gamma)]), pch=16, cex=1.5, col="blue")
  }
}
